rustsec/src/repository/git/modification_time.rs

```diff
@@ -31,69 +31,163 @@ impl GitModificationTimes {
                 };
-                let diff = repo.diff_tree_to_tree(prev_tree.as_ref(), Some(&tree), None)?;
-                for delta in diff.deltas() {
-                    let file_path = delta.new_file().path().unwrap();
-                    let file_mod_time = commit.time();
-                    mtimes
-                        .entry(file_path.to_owned())
-                        .and_modify(|t| *t = max(*t, file_mod_time))
-                        .or_insert(file_mod_time);
-                    ctimes
-                        .entry(file_path.to_owned())
-                        .and_modify(|t| *t = min(*t, file_mod_time))
-                        .or_insert(file_mod_time);
-                }
+
+                mtimes
+                    .entry(file_path.clone())
+                    .and_modify(|t| *t = max(*t, file_mod_time))
+                    .or_insert(file_mod_time);
+                ctimes
+                    .entry(file_path)
+                    .and_modify(|t| *t = min(*t, file_mod_time))
+                    .or_insert(file_mod_time);
             }
         }
-        Ok(GitModificationTimes { mtimes, ctimes })
+
+        Ok(Self { mtimes, ctimes })
     }
 
     /// Looks up the Git modification time for a given file path.
     /// The path must be relative to the root of the repository.
-    pub fn for_path(&self, path: GitPath<'_>) -> &Time {
-        self.mtimes.get(path.path()).unwrap()
+    pub fn for_path(&self, path: GitPath<'_>) -> Time {
+        *self.mtimes.get(path.path()).unwrap()
     }
 
     /// Looks up the Git creation time for a given file path.
     /// The path must be relative to the root of the repository.
     pub fn mdate_for_path(&self, path: GitPath<'_>) -> Date {
-        Date::from_str(&Self::git2_time_to_date(
-            self.mtimes.get(path.path()).unwrap(),
-        ))
-        .unwrap()
+        Self::gix_time_to_date(self.mtimes.get(path.path()).unwrap())
     }
 
     /// Looks up the Git creation time for a given file path.
     /// The path must be relative to the root of the repository.
     pub fn cdate_for_path(&self, path: GitPath<'_>) -> Date {
-        Date::from_str(&Self::git2_time_to_date(
-            self.ctimes.get(path.path()).unwrap(),
-        ))
-        .unwrap()
+        Self::gix_time_to_date(self.ctimes.get(path.path()).unwrap())
     }
 
-    fn git2_time_to_date(git_timestamp: &Time) -> String {
-        let unix_timestamp: u64 = git_timestamp.seconds().try_into().unwrap();
-        let duration_from_epoch = std::time::Duration::from_secs(unix_timestamp);
-        let mut time =
-            humantime::format_rfc3339(std::time::UNIX_EPOCH.add(duration_from_epoch)).to_string();
-        time.truncate(10);
-        time
+    fn gix_time_to_date(timestamp: &Time) -> Date {
+        let odt = crate::repository::git::gix_time_to_time(*timestamp);
+        let date = odt.date();
+
+        format!(
+            "{:0>4}-{:0>2}-{:0>2}",
+            date.year(),
```

Comment:
The time formatting stuff is annoying to use, this is just a manual version since the date part is trivial.